A rough custom element elm-pkg-js example

wolfadex opened the pull request:
I'd like to double check a few implementation details tomorrow, but this is roughly what a custom element wrapper elm-pkg-js would look like in my brief experience.

wolfadex committed:
A rough custom element elm-pkg-js example

wolfadex committed:
Switch from hybrids to FAST

FAST docs are a little bit better which I think will lend to better understanding and extension of the custom element portion.

wolfadex committed:
Add comments to better understand the code

wolfadex commented:
@supermario let me know if there's anything additional you'd like to see in this. It doesn't cover the entire Square Web Payments SDK, but does cover enough that you could take an actual card payment.

supermario commented on examples/elm-square-web-sdk/src/elm-square-web-sdk.js:
```diff
@@ -0,0 +1,76 @@
+
+  // This delay is to give time for the injected scripts
+  // to download and run. Not ideal, but it does work.
+  setTimeout(() => initSquareWebSdk(app, window.FAST), 1000);
```
This is unfortunate 🤔 does the Square SDK not have an onloaded event?

Is it possible to use the JS one instead? I.e.

```
squareWebSdk.onload = function() { initSquareWebSdk(app, window.FAST) };
document.head.appendChild(squareWebSdk);
...
```

wolfadex replied:
This never ocurred to me! I'll try it out quick and update accordingly

wolfadex replied:
I was able to switch to `onload` for the Square SDK, but not for FAST. Not sure if you have a recommendation for a custom element lib that doesn't use modules? I was having a hard time finding one that would work well for this example. I ask because `onload` doesn't work for the module script because it has to load and then add itself to the global scope, which causes a race condition with actually building the component.

supermario replied:
> Not sure if you have a recommendation for a custom element lib that doesn't use modules?

No, sorry :( I wasn't aware custom elements needed extra deps, is there a lot of complexity in doing a custom element by hand?

wolfadex replied:
It's not terrible by hand, I'll try and do that and see how safe I can make it. It can get kinda lengthy if you properly hand all the connecting and disconnecting. I'll try to have something by the end of the weekend.

supermario replied:
No rush @wolfadex!

supermario replied:
@wolfadex just stumbled through this again passing by – was this something you managed to look into any further? :)

supermario reviewed (approved):
LGTM! Minor comment.

wolfadex committed:
Prevent race conditions building component 

Use `onload` for the traditional script and call a global function from the module script.

supermario commented:
Merging for now – we can revisit later if needed :)
